SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -2175,23 +2145,29 @@ void USpatialNetDriver::AcceptNewPlayer(const FURL& InUrl, const FUniqueNetIdRep
 		UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Join failure: %s"), *ErrorMsg);
 		SpatialConnection->FlushNet(true);
 	}
+
+	// Preallocate the PlayerController entity so the AuthorityDelegation client authoritative components can be set
+	// correctly at spawn.
+	USpatialActorChannel* Channel = GetOrCreateSpatialActorChannel(SpatialConnection->PlayerController);
+	USpatialNetConnection* NetConnection = Cast<USpatialNetConnection>(Channel->Actor->GetNetConnection());
+	check(NetConnection != nullptr);
+	NetConnection->PlayerControllerEntity = Channel->GetEntityId();
 }
 
 // This function is called for server workers who received the PC over the wire
-void USpatialNetDriver::PostSpawnPlayerController(APlayerController* PlayerController, const FString& ClientWorkerId)
+void USpatialNetDriver::PostSpawnPlayerController(APlayerController* PlayerController)
 {
 	check(PlayerController != nullptr);
-	checkf(!ClientWorkerId.IsEmpty(), TEXT("A player controller entity must have an owning client worker ID."));
 
 	PlayerController->SetFlags(GetFlags() | RF_Transient);
 
 	FString URLString = FURL().ToString();
-	URLString += TEXT("?workerAttribute=") + ClientWorkerId;
 
 	// We create a connection here so that any code that searches for owning connection, etc on the server
 	// resolves ownership correctly
 	USpatialNetConnection* OwnershipConnection = nullptr;
-	if (!CreateSpatialNetConnection(FURL(nullptr, *URLString, TRAVEL_Absolute), FUniqueNetIdRepl(), FName(), &OwnershipConnection))
+	if (!CreateSpatialNetConnection(FURL(nullptr, *URLString, TRAVEL_Absolute), FUniqueNetIdRepl(), FName(),
+									SpatialConstants::INVALID_ENTITY_ID, &OwnershipConnection))
```

Comment:
Wouldn't this mean player controllers that get handed over won't be correctly cleaned up?

Reply:
Not sure what you mean here?

Reply:
If we're passing `INVALID_ENTITY_ID` here, this connection won't get added to `WorkerConnections`, so when `USpatialReceiver::OnRemoveEntity` gets called for the client worker entity, it won't clean up the corresponding connection.

Reply:
Yeah this is a bug - any PC that is handed over won't have it's hierarchy automatically deleted if the client quits. Tracking bug - https://improbableio.atlassian.net/browse/UNR-4492